### PR TITLE
Replace ngrok by sauce connection tunnel

### PR DESCRIPTION
**Pull request by denar90**
Fix for issue #7

### CODE REVIEW

**Comment by philipwalton** [comment]
Thanks for taking the time to look into this! I like what I'm seeing so far; I just have a couple of comments:

First, I think it'd be best to keep ngrok around as an option since sauce connect is quite a bit slower (and appears to be flakier as well). It also looks like ngrok raised the limit to 40 connections/minute, which should significantly improve things.

Maybe there should be a new `tunnel` option, that defaults to `ngrok` but can also be `sauce-connect` (and accept options). For this PR you can probably just add this as option in `lib/easy-sauce.js` (as you've done), and I can work out supporting them as command line options in the future.

Second (related to the flakiness I referenced above), I'm curious if you were able to get all your tests to pass. I'm unable to get my tests passing on iPhone/Safari 9.2 (it always fails), and I'm experiencing intermittent failures on other platforms. Are you seeing this too? (I know that's probably a Sauce Labs / Sauce Connect issue and unrelated to your changes.

**Comment by denar90** [comment]
Thank you for so well-grounded review. I'll add those changes. About tests, I tried on MacOS, it was working. I still have trial period on saucelabs, hence can share credentials. 
Moreover, would you like to have `tunnel` as command line option too?

**Comment by denar90** [comment]
Note: haven't tested new functionality with real connect. Hope to do it soon...

**Comment by denar90** [comment]
Finished with it.

File: README.md [hunk 1]
```diff
@@ -70,6 +70,10 @@ Options:
                     "jasmine", "qunit", "YUI Test", or "custom".
                     Defaults to "mocha".
                     See https://goo.gl/5KfjDS for details.
+                    
+  -T, --tunnel      Tunnel which will be created for running tests.
```
Comment (philipwalton) [inline README.md]: Can you make this "-s, --service"? I'd rather not have two options with the same letter if possible.

File: bin/usage.txt [hunk 2]
```diff
@@ -38,6 +38,10 @@ Options:
                     Defaults to "mocha".
                     See https://goo.gl/5KfjDS for details.
 
+  -T, --tunnel      Tunnel which will be created for running tests.
```
Comment (philipwalton) [inline bin/usage.txt]: Same comment as above.

File: lib/cli.js [hunk 3]
```diff
@@ -40,6 +40,7 @@ module.exports = function(argv) {
     if (argv.f || argv.framework) cliOpts.framework = argv.f || argv.framework;
     if (argv.v || argv.verbose) cliOpts.verbose = argv.v || argv.verbose;
     if (argv.q || argv.quiet) cliOpts.quiet = argv.q || argv.quiet;
+    if (argv.T || argv.tunnel) cliOpts.tunnel = argv.q || argv.tunnel;
```
Comment (philipwalton) [inline lib/cli.js]: `argv.q` should be `argv.T` (or `argv.s` once the option is changed).

**Review by philipwalton — COMMENTED** [review]
Thanks, I've made a few comments.

I'm still having some issues with Sauce Connect not being able to connect to the server, but I'll figure those out myself.

Once you fix the issues I pointed out I'll merge this PR.

**Comment by philipwalton** [comment]
Thanks! Merged.

**Pull request merged**